### ISSUE TITLE
PdoBackend for caches does not create the needed tables with mysql

### ISSUE THREAD

**Issue by jhinzmann** [neos/flow-development-collection#884]
### Description

The PDO-Backend creates the caching-tables only if sqlite is used.
When mysql is used, no tables are generated and the caching does not work.
If sqlite is used, the tables are generated from `TYPO3.Flow/Resources/Private/Cache/SQL/DDL.sql` by calling `PdoBackend->createCacheTables()` in `PdoBackend->connect()`

After creating the tables manually from the mentioned DDL.sql the caching does work.
The documentation does not mention this, therefore I guess it's a bug.

Maybe this should go in another issue, but when creating the tables manually, it's a good idea to not go with `text` (as in the DDL.sql) for `content` since the content of one cache entry can easily be longer than 64kb.
This is no problem in sqlite because there are no such limits on the columns.

### Steps to Reproduce

1. add a Caches.yaml with the following config:
```TYPO3_TypoScript_Content:
  backend: TYPO3\Flow\Cache\Backend\PdoBackend
  backendOptions:
    dataSourceName: 'mysql:host=XXX;dbname=XXX'
    username: 'XXX'
    password: 'XXX'`
```
2. Open the Frontend or Backend or try to flush the cache

#### Expected behavior

The caching tables should be created if needed and should be used for storing the caches

#### Actual behavior

The caching tables are not created.
The Frontend and Backend throw:
`SQLSTATE[42S02]: Base table or view not found: 1146 Table 'XXX.cache' doesn't exist`

Trying to flush the caches throws:
```
SQLSTATE[42S02]: Base table or view not found: 1146 Table 'XXX.tags' doesn't exist

  Type: PDOException
  Code: 42S02
  File: Packages/Framework/TYPO3.Flow/Classes/TYPO3/Flow/Cache/Backend/PdoBackend.php
  Line: 217
```
### Affected Versions

This bug occurred with
Neos: 2.3
Flow: 3.3
but by checking the PdoBackend implementation I guess this is still the case for Flow 4.0 and was the case since the PDO-Backend was introduced (https://github.com/neos/flow-development-collection/commit/92ad0c115da5dca736d9a1cf6cfd138d70f2403d#diff-d0f3ffadf243879866b8b49abddce315R345)

**Comment by aertmann** [comment]
Thanks for opening, been surprised by this too. Can't really be considered a bug though, but definitely a missing feature and an undocumented manual step.

**Comment by jhinzmann** [comment]
I've never contributed to Flow so far, but the steps to fix/implement this would be:

- [ ] create a pull-request with the feature implemented (checking and creating the tables) based on master
- [ ] create a pull-request for updating the documentation based on the lowest supported version which would be 3.2

Am I right?
Should there two separate issues for the bugfix and the feature?

**Comment by aertmann** [comment]
Yep, you're are spot on. No need for a separate issue.

**Comment by kdambekalns** [comment]
Finally fixed with #1383.
